osbs/build/spec.py

```diff
@@ -144,8 +147,11 @@ def set_params(self, git_uri=None, git_ref=None, git_branch=None, registry_uri=N
             raise OsbsValidationException("yum_repourls must be a list")
         self.yum_repourls.value = yum_repourls or []
         self.use_auth.value = use_auth
-        self.name.value = '%s-%s' % (self.component.value, self.git_branch.value)
-        self.base_image.value = os.path.join(registry_uri, base_image)
+        self.name.value = name_label.replace('/', '-')
+        self.trigger_imagestream_name.value = get_imagestream_name_from_image(base_image)
+        # imagestream and buildconfig have precisely the same name
+        self.imagestream_name.value = self.name.value
```

Comment:
I'm happy about this now. The `Name` label, with '/' replaced by '-', looks like the correct name to use for the ImageStream.